src/Ocelot/Authorization/ScopesAuthorizer.cs

```diff
@@ -28,14 +28,22 @@ public Response<bool> Authorize(ClaimsPrincipal claimsPrincipal, List<string> ro
                 return new ErrorResponse<bool>(values.Errors);
             }
 
-            var userScopes = values.Data;
+            IList<string> userScopes = values.Data;
 
-            var matchesScopes = routeAllowedScopes.Intersect(userScopes);
+            if (userScopes.Count == 1)
+            {
+                var scope = userScopes[0];
+
+                if (scope.Contains(' '))
+                {
+                    userScopes = scope.Split(' ', StringSplitOptions.RemoveEmptyEntries);
+                }
+            }
 
-            if (!matchesScopes.Any())
+            if (routeAllowedScopes.Except(userScopes).Any())

```

Comment:
This logic inversion is feasible, yet it appears redundant. It seems to be a minor refactoring aimed at reducing the number of lines in the code.
**Finally, it is useless change!**